pkg/controller/sparkapplication/controller.go

```diff
@@ -65,11 +69,24 @@ var (
 	execCommand = exec.Command
 )
 
+// AppQueueInfo is a struct to hold the queue and the last update timestamp for a SparkApplication
+type AppQueueInfo struct {
+	Queue        workqueue.RateLimitingInterface
+	LastUpdateTs time.Time
+}
+
+// A struck that will have map of queues and mutex for the map
+// We need many of those as one mutex will get us stuck in locks in high scale
+type AppQueuesMap struct {
+	appQueueMutex *sync.RWMutex
+	appQueueMap   map[string]AppQueueInfo
+}
+
```

Comment:
The naming of this struct is strange. The overall type name is `AppQueuesMap` but a field is named similarly `appQueueMap`. Can we give the struct type a more intuitive name? Something that indicates that it includes the mutex?

Reply:
Yes, it is not good naming, I will change it